### ISSUE TITLE
[Solved] Add ability to change query arguments

### ISSUE THREAD

**Issue by alexgarbarev** [Milad-Akarie/auto_route_library#1017]
Hello, and thanks for such wonderful library!

The problem I faced with - to dynamically change Query arguments. There is 2 use cases where I need that:

1. Add parallel navigation with Modal dialogs. For example we can add a query args like `?modal=create-doc&formId=123` and that will open a pop up with type `create-doc` and pass other args to it. And once dialog is closed - we can remove these args back.
That way we can share link to modal dialog and state is perfectly catched with the url.
2. To store current state inside URL, so we can share exact the state with single link. Somethinkg like `?mode=list` or `?mode=tree` that updated together with UI state.

So far I was able to achieve desired behavior and both cases works perfectly, but I had to fork and expose some of private API for that:
https://github.com/alexgarbarev/auto_route_library/commit/c0a46186904a9328484d120a0cdae2330ec6f29f

I'm not hoping that my changes could be merged, as I think author can make a more thoughtful and consistent API.

I had multiple attempts achieving that and updating query args that way - was the only way that worked constantly. I tried to use `navigateNamed` and `replaceNamed` finding current path, and adding query args, but that doesn't work with my complex flow (few nested and tabs routers).

My final solution is to update query args (see my commit) and then intercept that with Route Wrapper widget - it checks query args and does `showDialog` if needed.

**Comment by Milad-Akarie** [comment]
@alexgarbarev did you try navigating to the same path you're in with different query params
`router.navigateNamed("/create-doc?mode=tree");`
this should update the query params in both url and routeData.queryParams

**Comment by alexgarbarev** [comment]
Hi @Milad-Akarie yes, I tried that initially but faced with bugs when it navigated me to some another random screen. It worked good in sample app, but after I tried that in my real app (with tab based navigation, then few nested routers).

That's structure I had:

- /login
- /home (tab - based, with guard)
   - /home/tasks
   - /home/docs (nested routring)
      - /home/docs - list of docs
      - /home/docs/:id (second screen in nested navigation )
      
  So after I tried to change from `/home/docs/doc-1` to `/home/docs/doc-1?moda=tree` it redirects me to `/home/tasks`.
  Also I found that `navigationNamed` works randomly.. Sometimes it navigates to antoher screen, sometimes it doesn't update url, sometimes it doesn't remove params back (for example after I close the dialog, I need to remove params).. It's based on current state and current route and structure.
